### PR TITLE
Add/rename `braces`/`brackets`/`parenthesis` icons and improve `braces`/`json` metadata

### PR DESCRIPTION
**Pull request by danielbayley**

### CODE REVIEW

File: icons/parenthesis.svg
Comment (karsa-mistmere) [inline icons/parenthesis.svg:1]: This should probably be `parentheses` (with parenthesis as a tag)

File: icons/curly-braces.svg [hunk 1]
```diff
@@ -9,6 +9,6 @@
   stroke-linecap="round"
   stroke-linejoin="round"
 >
-  <path d="M8 3H7a2 2 0 0 0-2 2v5a2 2 0 0 1-2 2 2 2 0 0 1 2 2v5c0 1.1.9 2 2 2h1" />
-  <path d="M16 21h1a2 2 0 0 0 2-2v-5c0-1.1.9-2 2-2a2 2 0 0 1-2-2V5a2 2 0 0 0-2-2h-1" />
+  <path d="M8 5H7c-1.1 0-2 .9-2 2v3c0 1.1-.9 2-2 2 1.1 0 2 .9 2 2v3c0 1.1.9 2 2 2h1" />
+  <path d="M16 19h1c1.1 0 2-.9 2-2v-3c0-1.1.9-2 2-2-1.1 0-2-.9-2-2V7c0-1.1-.9-2-2-2h-1" />
```
Comment (jguddas) [inline icons/curly-braces.svg:13]:
```suggestion
  <path d="M8 5H7a2 2 0 0 0-2 2v3a2 2 0 0 1-2 2 2 2 0 0 1 2 2v3a2 2 0 0 0 2 2h1"/>
  <path d="M16 19h1a2 2 0 0 0 2-2v-3a2 2 0 0 1 2-2 2 2 0 0 1-2-2V7a2 2 0 0 0-2-2h-1"/>
```

File: icons/square-brackets.svg [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <polyline points="8,19 3,19 3,5.1 8,5.1" />
+  <polyline points="16,5.1 21,5.1 21,19 16,19" />
```
Comment (jguddas) [inline icons/square-brackets.svg:13]:
```suggestion
  <path d="M16 5.1h5V19h-5"/>
  <path d="M8 19H3V5.1h5"/>
```

**Review by ericfennis — APPROVED** [review]
Nice work, love these icons!

**Comment by karsa-mistmere** [comment]
I still think these are way too small.

Top row current, bottom row with this change, correlate these with most monospace fonts: https://fonts.google.com/?preview.text=%3C%7B%7D%3E&preview.text_type=custom&category=Monospace

image

Chevrons and braces usually have a ratio much closer to our current version.

**Comment by danielbayley** [comment]
> correlate these with most monospace fonts:
> …
> Chevrons and braces usually have a ratio much closer to our current version.

Some of those do have equal height though…

Since this is an icon font (nobody is going to be using Lucide for _actually coding_ (I hope! 😅) I’m not sure trying to match (most, but not all) existing monospace font proportions is a more important goal than having them look nice and consistent as a set…

@karsa-mistmere So are you pushing for the elongated versions, or saying just `curly-braces` should be, with the others smaller (or just chevrons) or some other combination…?

On a related note: should `chevrons-left-right` and `code` really be 2 separate icons?

**Comment by karsa-mistmere** [comment]
> I’m not sure trying to match (most, but not all) existing monospace font proportions is a more important goal than having them look nice and consistent as a set…

You know, that's exactly my point: I don't think they're going to look consistent at all if their ratios are utterly unlike 99.9% of real life examples. :)

I've done some research:

* [Fontawesome](https://fontawesome.com/search?q=code)
  image
* [Material Icons](https://fonts.google.com/icons)
  image
* [Codicons](https://microsoft.github.io/vscode-codicons/dist/codicon.html)
  image
* [Bootstrap icons](https://icons.getbootstrap.com/?q=code)
  image
* [Remix Icon](https://remixicon.com/)
  image
* [Phosphor Icons](https://phosphoricons.com/)
  image
* [Tabler icons](https://tablericons.com/)
  image
* [Unicons](https://iconscout.com/unicons/explore/line)
  image

Most have ratios pretty similar to what we currently have, and _in every single one of them_ the curly braces are taller if only by a little.

**Comment by karsa-mistmere** [comment]
> On a related note: should `chevrons-left-right` and `code` really be 2 separate icons?

Ouch. That's a tough nut, and something we should probably investigate further.

Maybe `chevrons-*` icons should match the base chevron size, but I'm not entirely sure. It might be more consistent, but it's maybe somewhat less balanced:
image

**Comment by jguddas** [comment]
> > On a related note: should chevrons-left-right and code really be 2 separate icons?

Should those be icons at all :wink:

File: icons/square-brackets.svg [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <polyline points="8 21 5 21 5 3 8 3" />
+  <polyline points="16 3 19 3 19 21 16 21" />
```
Comment (karsa-mistmere) [inline icons/square-brackets.svg:13]:
```suggestion
  <path d="M16 3h3v18h-3" />
  <path d="M8 21H5V3h3" />
```

**Comment by danielbayley** [comment]
@karsa-mistmere Ok, so just `push`ed those changes… You can see how they are modelled here:

Untitled

Untitled

The `chevrons-*`/`code` thing can be dealt with in a separate PR if necessary…

File: icons/parentheses.svg [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M8 21C2.7 16 2.7 8 8 3" />
+  <path d="M16 3c5.3 5 5.3 13 0 18" />
```
Comment (karsa-mistmere) [inline icons/parentheses.svg:13]: Just a minor pixel perfection fix.
```suggestion
  <path d="M8 21A12.13 12.13 0 0 1 8 3" />
  <path d="M16 3a12.13 12.13 0 0 1 0 18" />
```

File: icons/square-brackets.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  "$schema": "../icon.schema.json",
+  "tags": [
+    "code",
+    "token",
+    "array",
+    "list"
+  ],
+  "categories": [
+    "coding",
+    "development",
```
Comment (karsa-mistmere) [inline icons/square-brackets.json:11]:
```suggestion
    "development",
```

**Review by karsa-mistmere — CHANGES_REQUESTED** [review]

**Comment by danielbayley** [comment]
What do we think about this set being called simply `brackets`, `parentheses` and `braces` (with the latter being renamed from the current `curly-braces`, and `curly` moved to a `tag`)?

**Comment by karsa-mistmere** [comment]
> What do we think about this set being called simply `brackets`, `parentheses` and `braces` (with the latter being renamed from the current `curly-braces`, and `curly` moved to a `tag`)?

Yeah, that would definitely work for me, but let's wait for @ericfennis' input first.

---

And for now something completely different: why are you using force push? 🤔 

image

I keep running into the error above because of these, and I think it's generally not a good idea anyway, since with force pushes, we're losing some of the commit history and history about changes made.

It's not a super big deal per se, but it certainly makes it harder to review your pull requests, since by the time I get to some notifications, the respective changes will have already been overwritten, so if it's not absolutely necessary, I would very much appreciate it if you just used regular pushes. 😅

**Comment by karsa-mistmere** [comment]
We're force squashing them anyway, but while they're a work in progress, it's better to see all the revisions. :)
image

**Comment by danielbayley** [comment]
> We're force squashing them anyway, but while they're a work in progress, it's better to see all the revisions. :)

Ok, cool.

**Comment by danielbayley** [comment]
What do we think about these for `bracket-dot` and `bracket-error` (from codicons)… too much?

Untitled

**Comment by karsa-mistmere** [comment]
@danielbayley: Not totally sure, but in either case, `braces-dot` should use the same dot as `bell-dot` will, or the other way around.

**Comment by ericfennis** [comment]
@danielbayley @karsa-mistmere I'm ok with renaming them. These are the official names. But we maybe should aliases for backwards compatibility.

**Comment by danielbayley** [comment]
> @danielbayley @karsa-mistmere I'm ok with renaming them. These are the official names. But we maybe should aliases for backwards compatibility.

Done.

File: icons/parentheses.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    "parenthesis",
+    "parens",
+    "brackets",
+    "parameters",
+    "arguments",
+    "args",
+    "input",
+    "call",
+    "maths",
+    "formula"
```
Comment (karsa-mistmere) [inline icons/parentheses.json:15]:
```suggestion
    "formula",
    "function"
```

**Review by karsa-mistmere — APPROVED** [review]

**Review by ericfennis — APPROVED** [review]

**Pull request merged**